### PR TITLE
Added compatibility doctrine-bundle v2

### PR DESCRIPTION
**Pull request by Yartsev**
added-compatibility-doctrine-bundle2

### CODE REVIEW

**Comment by ruudk** [comment]
This is not mergeable. You remove all the typehints. Is there no other/better way?

**Comment by Yartsev** [comment]
Hi there. The class \Doctrine\Common\Persistence\ConnectionRegistry has changed its namespace https://github.com/doctrine/persistence/compare/v1.1.0...2.0.0 Now \Doctrine\Persistence\ConnectionRegistry
The class Doctrine\Common\Persistence\ManagerRegistry has been deleted  https://github.com/symfony/maker-bundle/issues/519#issuecomment-565728549 and we should use Symfony\Bridge\Doctrine\ManagerRegistry If we want to have support for new and old versions doctrine/persistence, we should do it. Suggest your solution? Thank you very much.

> This is not mergeable. You remove all the typehints. Is there no other/better way?

**Comment by cmodijk** [comment]
There are only 2 possible ways 

1. Type hint with both options in the PHPDoc block like this 
    ```php
    use Doctrine\Common\Persistence\ManagerRegistry as LegacyManagerRegistry;
    use Doctrine\Persistence\ManagerRegistry;

    /**
     * @var ManagerRegistry|LegacyManagerRegistry
     */
    ```
    *https://github.com/symfony/maker-bundle/pull/518/files#diff-e831369c31ef9da16227f0dfa61a163aR50*

 
1. Change the namespace to the new syntax and drop support dor the old version

**Comment by ruudk** [comment]
I think second option is best.

**Comment by Yartsev** [comment]
> I think second option is best.

If you choose "Change the namespace to the new syntax and drop support dor the old version" I can make a new commit with the transition to the new major version?

**Comment by ruudk** [comment]
Yes, please change it to the new FQCN. Also change the composer.json file to only allow `^2.0`.

**Comment by ruudk** [comment]
Can you amend your changes into 1 commit? Then I'm good.

File: src/Plugin/DoctrineORMPlugin/ResetClosedEntityManagers.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 namespace LongRunning\Plugin\DoctrineORMPlugin;
 
-use Doctrine\Common\Persistence\ManagerRegistry;
+use Symfony\Bridge\Doctrine\ManagerRegistry;
```
Comment (ruudk) [inline src/Plugin/DoctrineORMPlugin/ResetClosedEntityManagers.php]: I think this should be `\Doctrine\Persistence\ManagerRegistry` instead.

File: src/Plugin/DoctrineORMPlugin/ClearEntityManagers.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace LongRunning\Plugin\DoctrineORMPlugin;
 
-use Doctrine\Common\Persistence\ManagerRegistry;
+use Symfony\Bridge\Doctrine\ManagerRegistry;
```
Comment (ruudk) [inline src/Plugin/DoctrineORMPlugin/ClearEntityManagers.php]: I think this should be `\Doctrine\Persistence\ManagerRegistry` instead.

File: src/Plugin/DoctrineORMPlugin/ResetClosedEntityManagers.php (hunk 1 above)
Comment (Yartsev) [reply src/Plugin/DoctrineORMPlugin/ResetClosedEntityManagers.php]: That's my fault. You're right.

File: src/Plugin/DoctrineORMPlugin/ClearEntityManagers.php (hunk 2 above)
Comment (Yartsev) [reply src/Plugin/DoctrineORMPlugin/ClearEntityManagers.php]: That's my fault. You're right.

**Comment by cmodijk** [comment]
@ruudk Should we first release `1.0.0` of LongRunning first before merging this and then merge this and tag it as `2.0.0`? 

*https://github.com/LongRunning/LongRunning/tags*

**Comment by ruudk** [comment]
I don't think so. Composer will guard against issues. If the user of the library has DoctrineBundle 1.0 it will just refuse to update to later versions, and that is fine IMHO. So nothing should break.

People that are on 2.0 are able to install the newer version. And for them, nothing much will change either.

**Comment by cmodijk** [comment]
Hmm yes that is try until we need to fix something for the `1.0` version or are we not going to support that version?

**Comment by ruudk** [comment]
What I mean is that we don't have to make this project 1.0 at all. I would just bump the required Doctrine version and tag a new 0.x release.

**Comment by cmodijk** [comment]
Yes i know but if there is a bug in `LongRunning` than only the `doctrine/doctrine-bundle:^2.0` users will get the bug fixed.

**Comment by ruudk** [comment]
🤷‍♂️ then they should upgrade doctrine  too. Let's not over complicate the maintenance of this project. At least I'm not in for it.

**Comment by cmodijk** [comment]
Yeah i agree i was just wondering if we should

**Comment by Yartsev** [comment]
Sorry, so how will it be, will there be merge?

**Pull request merged**